query-engine/driver-adapters/src/lib.rs

```diff
@@ -1,17 +1,102 @@
 //! Query Engine Driver Adapters
-//! This crate is responsible for defining a quaint::Connector implementation that uses functions
-//! exposed by client connectors via N-API.
+//! This crate is responsible for defining a `quaint::Connector` implementation that uses functions
+//! exposed by client connectors via either `napi-rs` (on native targets) or `wasm_bindgen` / `js_sys` (on Wasm targets).
 //!
 //! A driver adapter is an object defined in javascript that uses a driver
-//! (ex. '@planetscale/database') to provide a similar implementation of that of a quaint Connector. i.e. the ability to query and execute SQL
-//! plus some transformation of types to adhere to what a quaint::Value expresses.
+//! (ex. '@planetscale/database') to provide a similar implementation of that of a `quaint::Connector`. i.e. the ability to query and execute SQL
+//! plus some transformation of types to adhere to what a `quaint::Value` expresses.
 //!
 
-mod async_js_function;
-mod conversion;
-mod error;
-mod proxy;
-mod queryable;
-mod result;
-mod transaction;
-pub use queryable::{from_napi, JsQueryable};
+pub(crate) mod conversion;
+pub(crate) mod error;
+pub(crate) mod proxy;
+pub(crate) mod queryable;
+pub(crate) mod send_future;
+pub(crate) mod transaction;
+pub(crate) mod types;
+
+pub use queryable::from_js;
+pub(crate) use transaction::JsTransaction;
+
+#[cfg(target_arch = "wasm32")]
+pub use wasm::JsObjectExtern as JsObject;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub use ::napi::JsObject;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub mod napi;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(crate) use napi::*;
+
+#[cfg(target_arch = "wasm32")]
+pub mod wasm;
+
+#[cfg(target_arch = "wasm32")]
+pub(crate) use wasm::*;
+
+#[cfg(target_arch = "wasm32")]
+mod arch {
+    use std::str::FromStr;
+
+    pub(crate) use super::JsObject;
+    pub(crate) use js_sys::JsString;
+    use js_sys::Reflect;
+    use tsify::Tsify;
+    use wasm_bindgen::JsValue;
```

Comment:
TODO: avoid the top-level module imports, and use the fully-specialized types in the function type signatures, for explicitness